public/app/features/explore/RichHistory/RichHistory.test.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { GrafanaTheme } from '@grafana/data';
+import { ExploreId } from '../../../types/explore';
+import { RichHistory, RichHistoryProps } from './RichHistory';
+import { Tabs } from './RichHistory';
+
+jest.mock('../state/selectors', () => ({ getExploreDatasources: jest.fn() }));
+
+const setup = (propOverrides?: Partial<RichHistoryProps>) => {
+  const props: RichHistoryProps = {
+    theme: {} as GrafanaTheme,
+    exploreId: ExploreId.left,
+    activeDatasourceInstance: 'Test datasource',
+    richHistory: [],
+    firstTab: Tabs.RichHistory,
+    deleteRichHistory: jest.fn(),
+  };
+
+  Object.assign(props, propOverrides);
+
+  const wrapper = mount(<RichHistory {...props} />);
+  return wrapper;
+};
+
+describe('RichHistory', () => {
+  it('should correctly render all tabs in tab bar', () => {
+    const wrapper = setup();
+    expect(wrapper.html()).toContain('Query history');
```

Comment:
Not sure I like the `contain` style of tests here. In a bigger component it may not tell us much about the rendering as the string can be anywhere in the html so this may pass even if it does not render the tabs. We should probably make it more narrow and for example check if there are 3 elements with class Tab and these titles.

Reply:
👍